src/CodeGen_PTX_Dev.cpp

```diff
@@ -791,6 +810,325 @@ bool CodeGen_PTX_Dev::supports_atomic_add(const Type &t) const {
 
 }  // namespace
 
+struct MatrixMultiplyInfo {
+    Expr A;
+    Expr B;
+    Expr C;
+
+    bool is_valid() const {
+        return A.defined() && B.defined() && C.defined();
+    }
+};
+
+MatrixMultiplyInfo is_matrix_multiply(const For *loop) {
+    if ((loop->for_type == ForType::Serial) && is_const_zero(loop->min) && is_const(loop->extent)) {
+        // The body of the loop can be a LetStmt or a Store
+        const LetStmt *let = loop->body.as<LetStmt>();
+        const Store *store = loop->body.as<Store>();
+
+        // If it is a let and the body is a store, use it
+        if (let && let->body.as<Store>()) {
+            store = let->body.as<Store>();
+        }
+
+        if (store) {
+            //  matmul$1[t26] = (float32)matmul$1[t26]
+            //      + (float32((float16)A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x])
+            //      * *float32((float16)B[(B.stride.1*matmul$1.s1.k$x) + t35]))
+            const Expr wild_f32x = Variable::make(Float(32), "*");
+            const Expr wild_f16x = Variable::make(Float(16), "*");
+            const Expr acc_pattern = wild_f32x + f32(wild_f16x) * f32(wild_f16x);
+            vector<Expr> matches;
+            if (expr_match(acc_pattern, store->value, matches)) {
```

Comment:
I think this whole set of ifs would be easier to read/understand if they were expressed as early returns:

    if (!expr_match(...)) {
        return MatrixMultiplyInfo();   
    }

    if (!load_c || !load_a || !load_b) {
        return MatrixMultiplyInfo();   
    }
...

Reply:
Done in 2c83da9ccfbbe54372a6a8b300c54f3d5994aa69